### PR TITLE
use xft::fused_multi_transformer_gpt in fused_multi_transformer_xpu_k…

### PR DESCRIPTION
**Pull request by NALLEIN**
### PR types

Performance optimization
### PR changes

OPs
### Description

Support fued_multi_transformer for multi_xpu inference, replaces gather and copy ops by gather_inplace, reuse max_data() in cache_kv xft_tensor. 
Releated beam_search_decode_op  [issue](https://github.com/PaddlePaddle/Paddle/issues/56611)

### CODE REVIEW

File: paddle/fluid/framework/ir/xpu/fused_multi_transformer_xpu_pass.cc [hunk 1]
```diff
@@ -528,6 +528,44 @@ int FusedMultiTransformerXPUPass::FusedMultiTransformerXPUQuant(
     cast_tofp32_func("FFN1Bias");
     cast_tofp32_func("FFN2Bias");
 
+    // Generate max_buffer: per_tensor_max and per_batch_max for kv_cache
+    int layer_num = fused_mt->Op()->Input("QKVW").size();
+    int max_ptr_size = GetXpuContext()->x_context()->max_ptr_size();
```
Comment (zhupengyang) [inline paddle/fluid/framework/ir/xpu/fused_multi_transformer_xpu_pass.cc]: https://github.com/PaddlePaddle/Paddle/blob/79bfb184dd6b529f2cb19867cd77628d37fad028/paddle/fluid/framework/ir/xpu/quant_utils.cc#L275
参考如上方式

File: paddle/phi/kernels/fusion/xpu/fused_multi_transformer_xpu_kernel.cc [hunk 2]
```diff
@@ -95,6 +97,17 @@ void FusedMultiTransformerXpuKernel(
                     phi::errors::PreconditionNotMet(
                         "Only support trans_qkvw == true at now."));
 
+  void* bkcl_context = nullptr;
+  if (ring_id >= 0) {
+#if defined(PADDLE_WITH_XPU_BKCL)
+    bkcl_context =
+        paddle::platform::BKCLCommContext::Instance().Get(ring_id)->comm();
+    // printf("ring id %d, bkcl_context %p\n", ring_id, bkcl_context);
```
Comment (zhupengyang) [inline paddle/phi/kernels/fusion/xpu/fused_multi_transformer_xpu_kernel.cc]: remove

File: paddle/phi/kernels/fusion/xpu/fused_multi_transformer_xpu_kernel.cc [hunk 3]
```diff
@@ -95,6 +97,17 @@ void FusedMultiTransformerXpuKernel(
...
+#else
+    printf("ring id %d, but no built with PADDLE_WITH_XPU_BKCL\n", ring_id);
```
Comment (zhupengyang) [inline paddle/phi/kernels/fusion/xpu/fused_multi_transformer_xpu_kernel.cc]: 用 glog 方式

**Comment by NALLEIN** [comment]
@chenwhql @qingqing01  能否帮忙approval一下CI的问题呀

**Review by chenwhql — DISMISSED** [review]
LGTM for const_cast

**Review by yuanlehome — DISMISSED** [review]

**Review by hong19860320 — APPROVED** [review]
LGTM

**Comment by mayang002** [comment]
LGTM

**Review by yuanlehome — APPROVED** [review]
LGTM for yaml change

**Review by heavengate — APPROVED** [review]

**Review by chenwhql — APPROVED** [review]

**Pull request merged**